src/WebJobs.Extensions.DurableTask/ManagedIdentityTokenSource.cs

```diff
@@ -51,25 +51,18 @@ public async Task<string> GetTokenAsync()
             DefaultAzureCredential defaultCredential;
             DefaultAzureCredentialOptions defaultAzureCredentialOptions = new DefaultAzureCredentialOptions();
 
-            if (this.Options != null && ((this.Options.AuthorityHost != null) || (!string.IsNullOrEmpty(this.Options.TenantId))))
+            if (this.Options?.AuthorityHost != null)
             {
-                if (this.Options.AuthorityHost != null)
-                {
-                    defaultAzureCredentialOptions.AuthorityHost = this.Options.AuthorityHost;
-                }
-
-                if (!string.IsNullOrEmpty(this.Options.TenantId))
-                {
-                    defaultAzureCredentialOptions.InteractiveBrowserTenantId = this.Options.TenantId;
-                }
-
-                defaultCredential = new DefaultAzureCredential(defaultAzureCredentialOptions);
+                defaultAzureCredentialOptions.AuthorityHost = this.Options.AuthorityHost;
             }
-            else
+
+            if (!string.IsNullOrEmpty(this.Options?.TenantId))
             {
-                defaultCredential = new DefaultAzureCredential();
+                defaultAzureCredentialOptions.InteractiveBrowserTenantId = this.Options.TenantId;
```

Comment:
I'm only noticing this for the first time now, but what affect does setting `InteractiveBrowserTenantId` have in a Durable HTTP context? The property name seems to imply the interactive use of a browser, which wouldn't apply here. I noticed there is also a [SharedTokenCacheTenantId](https://docs.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredentialoptions.sharedtokencachetenantid?view=azure-dotnet#Azure_Identity_DefaultAzureCredentialOptions_SharedTokenCacheTenantId) property. How is that one different from our perspective?

Reply:
In the linked issue #1279, a customer asked for access to set the `tenantId` through `AzureServiceTokenProvider` (the current method to get tokens). In that context, `tenantId` refers to the customer's AAD tenantId.

In the docs for `DefaultAzureCredentialOptions`, it says that `InteractiveBrowserTenantId` can be set through the `AZURE_TENANT_ID` environment variable. Since `AZURE_TENANT_ID` is equivalent to an AAD tenantId, it seems like the better option compared to `SharedTokenCacheTenantId`. I think the name `InteractiveBrowserTenantId` just isn't very clear.